packages/iTwinUI-react/src/core/Dialog/DialogMain.tsx

```diff
@@ -79,7 +84,7 @@ export const DialogMain = React.forwardRef<HTMLDivElement, DialogMainProps>(
         ref?.contains(document.activeElement) &&
           previousFocusedElement.current?.focus();
       };
-    }, [isOpen]);
+    }, [isOpen, setFocus]);
```

Comment:
Won't this unnecessarily trigger this entire effect when `setFocus` changes from `false` to `true`? We only want the effect to be triggered when `isOpen` changes, so I'd say put `setFocus` in a ref.

Reply:
And what if someone wants to set focus after dialog was opened?

Reply:
so if `setFocus` is set to true before the dialog is opened, then it should still set focus after the dialog opens

Reply:
No, I mean if dialog was opened and some other action happened that makes already opened dialog to be focused/
But still `setFocus` is boolean so it won't change on every rerender so I don't see a problem here.

Reply:
Why would anyone want to focus the dialog using the prop after it's already opened? It will also break the `previousFocusedElement` logic, so there's an even stronger reason to prevent this kind of behavior. Plus it will trigger this useEffect when setFocus changes even when the dialog is closed.

If the user really has a valid use case for this, then they should call `dialogRef.current.focus()`.

Reply:
I don't think it matters if having `setFocus` doesn't break anything. But if it's breaking something else and it can be achieved in another way then I'd say don't include `setFocus`. But generally I think it's good to enable users to have more options even if we don't see the point. I've talked to people that want to use/change our stuff in a way we didn't consider and sometimes that makes using our stuff more difficult than it needs to be.

Reply:
Changed to ref although I completely disagree with this...